### PR TITLE
Add `skipEmptyString` option

### LINKED ISSUE #244: Add parameter skipEmptyString
I love the skipNull that is available in the stringify.
Would it be posible to also include a skipEmpty parameter that filters all params where the value = ""

### PR DESCRIPTION
**Pull request by mum-never-proud**
Fixes #244

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -263,6 +269,14 @@ function parse(input, options) {
 	}, Object.create(null));
 }
 
+function isEmptyString(value) {
+	return value === '';
+}
```
Comment (sindresorhus) [inline index.js]: This utility has no value over just checking directly.

**Comment by sindresorhus** [comment]
You also need to update index.d.ts. That one and the readme should be in sync. I would also like to see more tests.

File: index.js (hunk 1 above)
Comment (mum-never-proud) [reply index.js]: done

File: index.js [hunk 2]
```diff
@@ -282,14 +292,18 @@ exports.stringify = (object, options) => {
...
 	const formatter = encoderForArrayFormat(options);
 
-	const objectCopy = Object.assign({}, object);
-	if (options.skipNull) {
-		for (const key of Object.keys(objectCopy)) {
-			if (objectCopy[key] === undefined || objectCopy[key] === null) {
-				delete objectCopy[key];
+	const objectCopy = Object
+		.keys(object)
+		.reduce((objectCopy, key) => {
```
Comment (sindresorhus) [inline index.js:302]: Don't do unrelated changes. I prefer this to stay a `for-of` loop instead of reduce, for readability.

File: readme.md [hunk 3]
```diff
@@ -304,6 +304,31 @@ queryString.stringify({a: undefined, b: null}, {
 //=> ''
 ```
 
+##### skipEmptyString
+
+Skip keys with empty strings as the value.
```
Comment (sindresorhus) [inline readme.md:309]:
```suggestion
Skip keys with an empty string as the value.
```

File: index.js (hunk 2 above)
Comment (mum-never-proud) [reply index.js:302]: I didn't refactor, rather a related change

```javascript
if (options.skipNull) {
  for (const key of Object.keys(objectCopy)) {
    if (objectCopy[key] === undefined || objectCopy[key] === null) {
      delete objectCopy[key];
    }
  }
}
```

the same can be achieved with 

```javascript
if (options.skipNull || options. skipEmptyString) {
  for (const key of Object.keys(objectCopy)) {
    if (objectCopy[key] === undefined || objectCopy[key] === null || objectCopy[key] === '') {
      delete objectCopy[key];
    }
  }
}
```
suppose in future if we introduce other filters the list will go on...

is there any better way than this?

Comment (sindresorhus) [reply index.js:302]: > is there any better way than this?

Do the same as now, but instead of reduce, use `for-of` loop, meaning no `Object.asssign`. You could maybe move the filtering logic here into a helper function (in this scope, not top-level).

Comment (mum-never-proud) [reply index.js:302]: hmm ok, if I remove the reduce this is how the code is (existing) will this work? sorry if I am not getting your point

```javascript
const objectCopy = Object.assign({}, object);

for (const key of Object.keys(object)) {
    if ((options.skipNull && isNullOrUndefined(object[key])) ||
        (options.skipEmptyString && object[key] === '')) {
        delete objectCopy[key];
    }
}
```

this is another approach

```javascript
const shouldFilter = key => (
    (options.skipNull && isNullOrUndefined(object[key])) ||
    (options.skipEmptyString && object[key] === '')
);

const formatter = encoderForArrayFormat(options);

const objectCopy = {};

for (const key of Object.keys(object)) {
    if (!shouldFilter(key)) {
        objectCopy[key] = object[key];
    }
}
```

we can add as many filters we need in future

your thoughts?

Comment (sindresorhus) [reply index.js:302]: ^ The last code block here is what I had in mind. 👌 Sorry for being unclear.
